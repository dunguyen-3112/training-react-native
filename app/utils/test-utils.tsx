import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { NavigationContainer } from '@react-navigation/native';

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return <NavigationContainer>{children}</NavigationContainer>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
