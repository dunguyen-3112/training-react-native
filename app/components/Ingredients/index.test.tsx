import React from 'react';
import { render, screen } from '@utils/test-utils';

import Ingredient from './index';
import { MOCK_INGREDIENTS } from '@__mock__';

describe('Test Ingredient', () => {
  test('should', async () => {
    const component = render(<Ingredient data={MOCK_INGREDIENTS} />);
    expect(component.toJSON()).toMatchSnapshot();

    const seeAll = await screen.findByText('See All');
    const title = await screen.findByText('Ingrediants');
    expect(seeAll).toBeTruthy();
    expect(title).toBeTruthy();
  });
});
