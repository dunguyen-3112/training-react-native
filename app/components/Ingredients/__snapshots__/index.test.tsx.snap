// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Ingredient should 1`] = `
<View
  style={
    Object {
      "marginTop": 4,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#5C5C5C",
              "fontSize": 20,
              "fontWeight": "600",
            },
            Object {},
          ]
        }
      >
        Ingrediants
      </Text>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#91C788",
              "fontSize": 11,
              "fontWeight": "500",
            },
            Object {},
          ]
        }
      >
        See All
      </Text>
    </View>
    <View />
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "name": "Bread",
          "value": 200,
        },
        Object {
          "id": 2,
          "name": "Meat (Chicken)",
          "value": 200,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#5C5C5C",
                  "fontSize": 15,
                  "fontWeight": "600",
                },
                Object {
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            Bread
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#5C5C5C",
                  "fontSize": 15,
                  "fontWeight": "600",
                },
                Object {},
              ]
            }
          >
            200 cal
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#5C5C5C",
                  "fontSize": 15,
                  "fontWeight": "600",
                },
                Object {
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            Meat (Chicken)
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#5C5C5C",
                  "fontSize": 15,
                  "fontWeight": "600",
                },
                Object {},
              ]
            }
          >
            200 cal
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
